packages/react/src/exampleTypes/StepSlider.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { FC, useState } from 'react'
+import _ from 'lodash/fp.js'
+import {
+  RangeSlider,
+  RangeSliderTrack,
+  RangeSliderFilledTrack,
+  RangeSliderThumb,
+  Box,
+  Tooltip,
+  Text,
+  HStack,
+  Stack,
+} from '@chakra-ui/react'
+import { contexturify } from '../utils/hoc'
+import { toNumber } from '../utils/format'
+import { FormatterFn, Node, Tree } from '../types/contexture'
+
+/** This theme is an initial step towards looking further
+ * at the tokenization to be set by our design system. */
+const theme = {
+  color: {
+    blue: { 100: '#CCE7FF', 500: '#0076DE', 700: '#005199' },
+    gray: { 50: '#FFFFFF', 100: '#E7EAEE' },
+    container: { border: { sm: '#E7EAEE' } },
+    text: {
+      body: {
+        primary: '#495569',
+        tertiary: '#697996',
+      },
+    },
+  },
+  radius: { base: { xs: '8px' } },
+  font: { size: { xs: '10px', md: '14px' } },
+}
+
+export type UnknownRec = Record<string, unknown>
+
+type NodeProps = {
+  range: number[]
+  steps: number[]
+}
+
+type ContextProps = {
+  recordsCount: number
+}
+
+interface StepSliderProps {
+  node: Node<NodeProps, ContextProps>
+  formatter: FormatterFn
+  path: string
+  tree: Tree<NodeProps>
+}
+
+const getDefaultRange = (steps: number[], range: number[]): number[] => {
+  if (range.length === 0) return [0, steps.length - 1]
+  const minStep = steps.findIndex((step: number) => step === range[0])
+  const maxStep = steps.findIndex((step: number) => step === range[1])
+  return [minStep, maxStep]
+}
+
+const getRealValue = (steps: number[], value: number): number => {
+  return steps[value]
+}
+
+const StepSlider: FC<StepSliderProps> = ({
+  tree,
+  node,
+  formatter = toNumber,
+}) => {
+  const formatValue = (value: number): string => {
+    const realValue = getRealValue(steps, value)
+    const lastStepValue = steps[steps.length - 1]
+    if (realValue === lastStepValue) return `${formatter(realValue)}+`
+    return formatter(realValue).toString()
+  }
+
+  const steps = node.steps
+  const count = node.context.recordsCount ?? 0
+  const defaultRange = getDefaultRange(steps, node.range)
+  const [rangeValues, setRangeValues] = useState(defaultRange)
```

Comment:
```suggestion
  const [rangeValues, setRangeValues] = useState<number[]>(defaultRange)
```

Reply:
Type inference is doing that job for us but I will add it so it makes more clear:
![Screenshot 2024-08-19 at 10 47 37](https://github.com/user-attachments/assets/b10f0cce-bc5d-4a93-beda-c2d687a7d522)